scene/2d/animated_sprite_2d.cpp

```diff
@@ -217,14 +217,21 @@ void AnimatedSprite2D::_notification(int p_what) {
 						}
 					}
 
+					timeout = _get_frame_duration();
+
 					queue_redraw();
 
 					emit_signal(SceneStringNames::get_singleton()->frame_changed);
 				}
 
-				double to_process = MIN(timeout, remaining);
+				double to_process = MIN(timeout / Math::abs(speed_scale), remaining);
+				timeout -= to_process * Math::abs(speed_scale);
```

Comment:
`abs()` is done above, so it is no longer needed here.

Reply:
I think this is more correct handling, as it doesn't allow either `remaining` or `timeout` to go below zero. Alternatively, we can replace `timeout = _get_frame_duration();` to `timeout += _get_frame_duration();`, but I haven't considered this option in detail, so I'm not sure if this is correct either.

Reply:
Hmm, okay. I recommend caching it to avoid the need to do the abs twice.
```cpp
float abs_speed_scale = Math::abs(speed_scale);
```

Reply:
Done.